packages/pluggableWidgets/language-selector-web/src/LanguageSelector.tsx

```diff
@@ -0,0 +1,59 @@
+import { ReactNode, createElement, useEffect, useState, useCallback } from "react";
+import { LanguageSelectorContainerProps } from "typings/LanguageSelectorProps";
+import { LanguageSwitcher } from "./components/LanguageSwitcher";
+import "./ui/LanguageSelector.scss";
+
+export type LanguageItem = {
+    _guid: string;
+    value: string;
+};
+
+export default function LanguageSelector(props: LanguageSelectorContainerProps): ReactNode {
+    const [selectedLanguage, setSelectedLanguage] = useState<LanguageItem | undefined>();
+    const [languageList, setLanguageList] = useState<LanguageItem[]>([]);
+    const [hideWidget, setHideWidget] = useState(false);
+
+    useEffect(() => {
+        if (props.languageOptions.items && props.languageCaption) {
+            const languages = props.languageOptions.items.map(item => ({
+                _guid: item.id,
+                value: props.languageCaption.get(item).value as string
+            }));
+            setLanguageList(languages);
+            if (languages.length < 2 && props.hideForSingle) {
+                setHideWidget(true);
+            }
+        }
+    }, [props]);
```

Comment:
Isn't it an equivalent of skipped props array? I think props are new on every re-render, so this effectively means - rerun on every re-render.

Reply:
Actually no, if you pass empty array it will trigger only once, if you don't pass anything it will render on every change, and only with props we have control over rendering cycle.

Reply:
Ah, true! Good point!